build_tools/check_backticks.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3 -u
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+"""Test script to check for invalid use of single-backticks."""
+
+__author__ = ["geetu040"]
+
+import argparse
+import ast
+import glob
+import re
+
+
+def find_py_files(folder_path):
+    """Find all Python files in a given folder path."""
+    return glob.glob(f"{folder_path}/**/*.py", recursive=True)
+
+
+def extract_docstrings(filename):
+    """Extract docstrings from a Python file."""
+    # create abstract syntax tree from the file
+    with open(filename) as f:
+        tree = ast.parse(f.read())
+
+    # walk through all nodes in the tree
+    docstrings = {}
+    for node in ast.walk(tree):
+        if (
+            isinstance(node, ast.Expr)
+            and isinstance(node.value, ast.Constant)
+            and isinstance(node.value.value, str)
+        ):
+            # if the node is an expression and
+            # its value is a constant and
+            # constant's value is a string
+            # the node represents a docstring
+            # See https://docs.python.org/3/library/ast.html#abstract-grammar
+            docstring = node.value.value
+            lineno = node.value.lineno
+            docstrings[lineno] = docstring
+
+    return docstrings
+
+
+def find_invalid_backtick_text(docstring):
+    """Find invalid backtick text in a docstring."""
+    # remove all the code blocks to avoid interference
+    # they are allowed to use backticks in any way
+    docstring = re.sub(r"```.*?```", "", docstring, flags=re.DOTALL)
+
+    # remove all double-backticks to avoid interference
+    # we are looking only for invalid single-backtick
+    docstring = re.sub(r"``.*?``", "", docstring)
```

Comment:
These two probably can be merged together, if 2 or more backticks are acceptable. Something like ```r"(`{2,}).+?\1"```, but it's not tested.

Reply:
I did this for the sake of simplicity.

The regex expressions can be merged but they become complex and unreadable when we try to avoid this situation: "regex 2 basically covers all the cases of regex 1 because it is a sub-part of it and thus the complete code-block is not substituted"

Reply:
I don't think I understood what you meant, but if backreference of captured group seems complex, you can keep as it is. It was only a suggestion.

Reply:
I was not using the correct `flags`, so I figured that out